### PR TITLE
D3ASIM-3447: Port client connections interfaces to d3a-interface

### PR DESCRIPTION
**Pull request by Hassan754**

### CODE REVIEW

File: d3a_interface/client_connections/websocket_connection.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+from d3a_interface.client_connections.utils import retrieve_jwt_key_from_server
+
+
+class WebsocketMessageReceiver(ABC):
+    def __init__(self, rest_client):
+        self.client = rest_client
+
+    @abstractmethod
+    def received_message(self, message):
+        ...
```
Comment (spyrostz) [inline d3a_interface/client_connections/websocket_connection.py:24]: Out of curiosity, is this preferred over the old-fashioned `pass` statement ?

**Review by spyrostz — APPROVED** [review]
LGTM, probably needs to be merged together with the api client PR

**Review by hannesdiedrich — APPROVED** [review]
LGTM

File: d3a_interface/client_connections/websocket_connection.py (hunk 1 above)
Comment (Hassan754) [reply d3a_interface/client_connections/websocket_connection.py:24]: It looks it is only a matter of preference, there are no rules about this in the PEPs

**Review by faizan2590 — APPROVED** [review]

File: d3a_interface/client_connections/websocket_connection.py (hunk 1 above)
Comment (fievelk) [reply d3a_interface/client_connections/websocket_connection.py:24]: If that's ok for you, I would then use `pass` (as it's used all across our codebase and it's more common I think).

Comment (Hassan754) [reply d3a_interface/client_connections/websocket_connection.py:24]: @fievelk Yea I'm okay with both options, I'll leave this here though if it's fine

**Pull request merged**